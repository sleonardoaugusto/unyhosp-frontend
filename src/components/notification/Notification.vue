<template>
  <div class="notification-bar">
    <p class="notification-bar__message">
      <span class="notification-bar__close" @click="close">x</span>
      {{ notification.message }}
    </p>
  </div>
</template>

<script>
export default {
  name: 'Notification',
  props: {
    notification: {
      type: Object,
      required: true
    }
  },
  methods: {
    close() {
      this.$emit('close-notification')
    }
  }
}
</script>

<style scoped>

</style>