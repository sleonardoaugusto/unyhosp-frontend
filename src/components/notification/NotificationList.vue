<template>
  <div class="notification-container">
    <Notification
      v-for="n in notifications"
      :notification="n"
      :key="n.id"
      @close-notification="close(n)"
    />
  </div>
</template>

<script>
import Notification from '@/components/notification/Notification'
import { mapActions, mapState } from 'vuex'

export default {
  name: 'NotificationList',
  components: {
    Notification
  },
  methods: {
    close(notification) {
      this.remove(notification)
    },
    ...mapActions({
      remove: 'notification/remove'
    })
  },
  computed:
    mapState({
      notifications: state => state.notification.notifications
    })
}
</script>

<style scoped>

</style>