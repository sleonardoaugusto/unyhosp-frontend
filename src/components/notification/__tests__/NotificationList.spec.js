import { createLocalVue, shallowMount } from '@vue/test-utils'
import Vuex from 'vuex'
import NotificationList from '@/components/notification/NotificationList'
import Notification from '@/components/notification/Notification'

const localVue = createLocalVue()

localVue.use(Vuex)

describe('NotificationList', () => {
  let wrapper

  let store
  let state
  let actions

  beforeEach(() => {
    state = {
      notifications: [{
        message: 'Problema de conexão: Network Error',
        type: 'error'
      }]
    }

    actions = {
      'notification/remove': jest.fn()
    }

    store = new Vuex.Store({
      modules: {
        notification: {
          state,
          actions
        }
      }
    })

    wrapper = shallowMount(NotificationList, {
      localVue,
      store
    })
  })

  it('renders the component', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders notification child component if $store notification is not empty', () => {
    const notification = wrapper.find(Notification)
    expect(notification.exists()).toBeTruthy()
  })

  it('validate binded notification computed prop passed to notification child component', () => {
    const notification = wrapper.find(Notification)
    expect(notification.vm.notification).toEqual(wrapper.vm.notifications[0])
  })

  it('calls close method at close-notification', () => {
    const closeFunction = spyOn(wrapper.vm, 'close')
    wrapper.find(Notification).vm.$emit('close-notification')
    expect(closeFunction).toHaveBeenCalled()
  })

  it('close method trigger store remove method', () => {
    const notificationData = {
      notification: {
        message: 'Problema de conexão: Network Error',
        type: 'error'
      }
    }
    expect(actions['notification/remove']).not.toHaveBeenCalled()
    wrapper.vm.close(notificationData)
    expect(actions['notification/remove']).toHaveBeenCalled()
  })
})