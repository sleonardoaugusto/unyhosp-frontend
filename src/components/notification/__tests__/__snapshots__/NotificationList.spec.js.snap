// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationList renders the component 1`] = `
<div class="notification-container">
  <notification-stub notification="[object Object]"></notification-stub>
</div>
`;
